File: Difficulty: Medium/Minimum Platforms/minimum-platforms.cpp
class Solution {
  public:
    int minPlatform(vector<int>& arr, vector<int>& dep) {
        // code here
        sort(arr.begin(),arr.end());
        sort(dep.begin(),dep.end());
        int i=0;
        int j=0;
        int ans=0;
        int temp=0;
        
        while(i<arr.size() && j<dep.size()){
            if(arr[i]<=dep[j]){
                i++;
                temp++;
                ans=max(ans,temp);
            }else{
                j++;
                temp--;
            }
        }
        return ans;
    }
};

File: Difficulty: Medium/Minimum Platforms/minimum-platforms.java
class Solution {
    public int minPlatform(int arr[], int dep[]) {
        //  code here
        int ans = 0;
        int j=0;
        int i=0;
        int count = 0;
        Arrays.sort(arr);
        Arrays.sort(dep);
        
        while(i<arr.length && j<arr.length){
            if(arr[i]<=dep[j]){
                count++;
                ans=Math.max(ans,count);
                i++;
            } else{
                count--;
                j++;
            }
        }
        return ans;
    }
}
